rama-http-backend/src/client/proxy/layer/proxy_connector/service.rs

```diff
@@ -182,6 +194,9 @@ where
         }
 
         let mut connector = InnerHttpProxyConnector::new(transport_ctx.authority.clone())?;
+        self.version.inspect(|version| {
```

Comment:
This logic will need to be different based on my other comment, as otherwise you'll miss the explicit None == auto choice.